### PR TITLE
31 add unit tests

### PR DESCRIPTION
**Pull request by DandyLyons**
https://github.com/brittanyarima/Steps/issues/31

Here are some unit tests. More to come.

### CODE REVIEW

**Comment by brittanyarima** [comment]
@DandyLyons thank you!! I'll be looking thru pull requests this evening. So excited about this one!

**Comment by brittanyarima** [comment]
@DandyLyons I'm getting this error _Value of type 'DependencyValues' has no member 'contributors'_ On ContributorsViewModel so this won't compile on my machine. Everything else looks fine though!

**Comment by DandyLyons** [comment]
> @DandyLyons I'm getting this error _Value of type 'DependencyValues' has no member 'contributors'_ On ContributorsViewModel so this won't compile on my machine. Everything else looks fine though!

It looks like ContributorsClient.swift did not get committed. I'm not sure how this happened. Working on it now.

**Comment by DandyLyons** [comment]
Ok on further research, it looks like it was ignored because `.gitignore` includes `Dependencies/`. I think we can safely remove this from `.gitignore` since we are not using Accio Dependency management.

**Comment by DandyLyons** [comment]
@brittanyarima I discovered an issue with using `git rm --cached <filename>`. It does remove that file from the commit history, which then does allow it to be respected by `.gitIgnore`. However, it also registers as a deleted file, which will then delete that file on anyone else's machine, which is not what we want. (We want to ignore it, remove it from the commit history, but keep it on each person's respective machine). 

Long story short, it seems like it's complicated/tricky to add a file to .gitIgnore if it has already been tracked. Here's an approach we might consider suggested by ChatGPT. (Very long message thread. The very last message is the relevant one.)

Ultimately, this .gitIgnore/ .xcconfig thing is getting to be so complicated, that it might need to be it's own Issue. 😬

**Comment by brittanyarima** [comment]
@DandyLyons Good point! Would you mind making an issue with the information you've
found? Maybe someone has dealt with this before and might be able to help!

**Comment by brittanyarima** [comment]
@DandyLyons I updated the .gitIgnore and removed the Dependencies/ file.

**Review by brittanyarima — CHANGES_REQUESTED** [review]
Just need that ContributorsClient file :)

**Comment by DandyLyons** [comment]
```
//
//  ContributorsClient.swift
//  Steps
//
//  Created by Daniel Lyons on 10/6/23.
//

import Foundation
import Dependencies

struct ContributorsClient {
    
    /// A closure that asynchronously fetches a Contributors
    var fetchContributors: () async -> [Contributor]
}

// MARK: Live Dependency Key
extension ContributorsClient: DependencyKey {
    
    /// A live value of `ContributorsClient` for
    static let liveValue = ContributorsClient(
        fetchContributors: {
            guard let url = URL(string: "https://api.github.com/repos/brittanyarima/Steps/contributors") else { return [] }
            
            do {
                let (data, _) = try await URLSession.shared.data(from: url)
                let contributors = try JSONDecoder().decode([Contributor].self, from: data)
                
                return contributors
            } catch {
                print("❗️ Error laoding contributors from Github API: \(error.localizedDescription)")
                return []
            }
        }
    )
}



// MARK: Test Dependency Key
extension ContributorsClient: TestDependencyKey {
    /// A preview instance of `ContributorsClient` which by default will be used in Xcode Previews
    static let previewValue = ContributorsClient.liveValue
    // It should be safe to use the live value in previews for now since the client can only fetch, not write.
    
    /// A test instance of `ContributorsClient` which by default will be used in XCTests
    static let testValue: Self = .init(
        fetchContributors: {
            return [Contributor].mock
        }
    )
    
}

// MARK: Dependency Values
extension DependencyValues {
    var contributors: ContributorsClient {
        get { self[ContributorsClient.self] }
        set { self[ContributorsClient.self] = newValue }
    }
}


```

**Review by brittanyarima — APPROVED** [review]
